lib/nats.js

```diff
@@ -1534,29 +1545,33 @@ Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
         msg = msg === undefined ? null : msg;
     }
 
+    if (opt_callback && typeof opt_callback !== 'function') {
+        throw (new NatsError(BAD_CALLBACK_MSG, BAD_CALLBACK));
+    }
+
     if (!subject) {
         if (opt_callback) {
             opt_callback(new NatsError(BAD_SUBJECT_MSG, BAD_SUBJECT));
+            return;
         } else {
             throw (new NatsError(BAD_SUBJECT_MSG, BAD_SUBJECT));
         }
     }
     if (typeof msg === 'function') {
-        if (opt_callback || opt_reply) {
+        if (opt_callback) {
```

Comment:
This check was broken: The code checks for the presence of opt_reply OR opt_callback, but then goes on to call `opt_callback` which of course fails when only `opt_reply` is present.